pkg/parser/parser.go

```diff
@@ -373,23 +375,95 @@ func (v *StmtVisitor) VisitStmt(s ast.Stmt) {
 	}
 }
 
-// findState check the change line
-// if it hits the statement, means change on that statement, and return "Changed"
-// otherwise, return "Original"
-func findState(stmt *StmtExtent, change *gittool.Change) State {
+// setStatementsState sets statements' State field according to the file change.
+// If change is nil, means no change at, it runs in full coverage mode
+// If change is not nil, loop over each changed lines and find its statement and set the statement to Changed
+func (parser *Parser) setStatementsState(change *gittool.Change, statements []*statement) {
 	if change == nil {
-		return Original
+		return
+	}
+	if len(statements) == 0 {
+		return
 	}
 
+	parser.logger.Debugf("processing changed file: %s", change.FileName)
 	for _, s := range change.Sections {
-		for i := s.StartLine; i <= s.EndLine; i++ {
-			if stmt.startLine <= i && stmt.endLine >= i {
-				return Changed
+		for lineNum := s.StartLine; lineNum <= s.EndLine; lineNum++ {
+			if isCodeLine(s.Contents[lineNum-s.StartLine]) {
```

Comment:
add ut for the isCodeLine?

Reply:
Covered in `TestSetStatementsState` cases.